employees/tests/test_views.py

```diff
@@ -862,3 +863,70 @@ def test_view_should_display_author_reports_only_one_project(self):
         self.assertNotContains(response, report_from_other_project.description)
         self.assertEqual(response.status_code, 200)
         self.assertTemplateUsed(response, AuthorReportProjectView.template_name)
+
+
+class ReportSummaryTests(TestCase):
+    def setUp(self):
+        self.user = ManagerUserFactory()
+        self.current_time = timezone.now()
+        self.work_hours_per_project = self._fill_database_with_user_reports(self.user, self.current_time)
+        self.total_monthly_hours = sum(self.work_hours_per_project.values(), timezone.timedelta())
+
+    def test_report_list_should_accurately_evaluate_work_hours_statistics(self):
+        self.client.force_login(self.user)
+        url = reverse("custom-report-list", kwargs={"year": self.current_time.year, "month": self.current_time.month})
+
+        context_data = self.client.get(url).context_data
+        self._assert_projects_work_percentage_is_evaluated_accurately(context_data)
+
+    def test_author_report_list_should_accurately_evaluate_work_hours_statistics(self):
+        admin_user = AdminUserFactory()
+        self.client.force_login(admin_user)
+        url = reverse(
+            "author-report-list",
+            kwargs={"pk": self.user.pk, "year": self.current_time.year, "month": self.current_time.month},
+        )
+
+        context_data = self.client.get(url).context_data
+
+        self._assert_projects_work_percentage_is_evaluated_accurately(context_data)
+
+    def _assert_projects_work_percentage_is_evaluated_accurately(self, context_data):
+        self.assertTrue("projects_work_percentage" in context_data.keys())
+
+        projects_work_percentage = context_data["projects_work_percentage"]
+
+        expected_result = {
+            project: ([work_hours_sum, work_hours_sum / self.total_monthly_hours * 100])
+            for project, work_hours_sum in self.work_hours_per_project.items()
+        }
+
+        self.assertEqual(expected_result, projects_work_percentage)
+
+    @staticmethod
+    def _fill_database_with_user_reports(user, base_date):
+        projects = []
+        work_hours_sum_for_current_month = {}
+        for i in range(4):
```

Comment:
I absolutely agree that properly descriptive variable names are a life saver, but to me it is clear that *i* stands for *index*. It's a widespread naming convention in every single programming language I know whenever using a *for* loop.
If our sensei thinks otherwise I will gladly challenge him to a Kendo match to settle it! I shall defend widespread and convenient 101-level conventions at all cost! 😛

Reply:
I mean, if I change it here and only because it is accessed inside the loop, shouldn't I change it in all other loops for consistency? That would result in needlessly descriptive code, which I believe would make it harder to read, completely beating the purpose.

Reply:
I mean you can just name it `index` instead of `i` or `list_name_index` to be precise :)

Reply:
Not related to this PR anymore